app/Check/Consumers/Client/ClientBuilder.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\Monitoring\Check\Consumers\Client;
+
+class ClientBuilder
+{
+
+	private ?int $connectTimeout;
+
+	private ?int $timeout;
+
+	/**
+	 * @see https://docs.guzzlephp.org/en/stable/request-options.html#verify
+	 */
+	private bool $verify = TRUE;
+
+	private ?\Pd\Monitoring\Check\Consumers\Client\Conf\AllowRedirectsInterface $allowRedirects = NULL;
+
+
+	private function __construct(
+		?int $connectTimeout,
+		?int $timeout
+	)
+	{
+		$this->connectTimeout = $connectTimeout;
+		$this->timeout = $timeout;
+	}
+
+
+	public function withVerify(bool $verify): self
+	{
+		$factory = clone $this;
```

Comment:
Předpokládám, že pomocí toho klonováni chceš zaručit neměnnost toho objektu. Misto clone bych radši udělal novou instanci takto
```suggestion
		return new self (
		    $this->connectTimeout,
		    $this->timeout,
		    $verify,
		    $this->allowRedirects
		);
```
Příjde mi zvláštní, že všechny parametry toho objektu není možné nastavit přímo v konstruktoru. Ja bych je tam dal, klidně s výchozíma hodnotama a pak bys mohl vytvářet nové instance přes new self().